### ISSUE TITLE
Electrode container project - build failure with carthage

### ISSUE THREAD

**Issue by anilmanukonda** [electrode-io/electrode-native#575]
Build failing on **ElectrodeContainer.xcodeproj** everytime when built with carthage.
`carthage bootstrap --platform ios`

**Error log**

`The following build commands failed:
	CompileC /Users/a6001805/Library/Caches/org.carthage.CarthageKit/DerivedData/9.2_9C40b/priceblockcontainer/v1.0.0/Build/Intermediates.noindex/ElectrodeContainer.build/Release-iphonesimulator/ElectrodeContainer.build/Objects-normal/x86_64/NSBundle+frameworkBundle.o ElectrodeContainer/NSBundle+frameworkBundle.m normal x86_64 objective-c com.apple.compilers.llvm.clang.1_0.compiler
(1 failure)`

**Comment by clairewli** [comment]
@anilmanukonda Thanks for reporting the issue. I'm wondering if you manually import ElectrodeContainer.xcodeproj into your project. Would it build for you?

**Comment by anilmanukonda** [comment]
Container project is not building independently either. PFA screenshot, most of the errors seems to be missing system frameworks like Foundation, Darwin
<img width="266" alt="container_build_errors" src="https://user-images.githubusercontent.com/8689237/36180853-ef18d2fe-10d6-11e8-8cd9-757410de9f98.png">

Also I got this error on react packager when building container project from xcode
`module.js:487
    throw err;
    ^

Error: Cannot find module '/Users/a6001805/Hybrid/ElectrodeSample/ElectrodeNativeAppSample/NativeSample/Carthage/Checkouts/priceblockcontainer/ElectrodeContainer/Libraries/ReactNative/local-cli/cli.js'
    at Function.Module._resolveFilename (module.js:485:15)
    at Function.Module._load (module.js:437:25)
    at Function.Module.runMain (module.js:605:10)
    at startup (bootstrap_node.js:158:16)
    at bootstrap_node.js:575:3
Process terminated. Press <enter> to close the window`

**Comment by clairewli** [comment]
@anilmanukonda We've run into this issue before. It's related to how the third-party run script inside React project. Based on past experience dealing with this, there are two ways people have to get around with it (it's a known issue from React and it's not fixed by them yet). 
1. `rm -rf ~/.ern/containergen/out/ios/ElectrodeContainer/Libraries/ReactNative/third-party/`
2. in your root directory of your Miniapp, `rm -rf ios` and then run `ern run-ios`.

**Comment by clairewli** [comment]
Can I ask which version of React Native you are using?

**Comment by belemaire** [comment]
The issue Claire is referring to is https://github.com/electrode-io/electrode-native/issues/226
We still haven't found a clean resolution for this one, the manual approach mentioned by Claire (and present in the original issue comments as well) works somehow.

**Comment by jomnius** [comment]
I've had similar problems many times, thanks for a new tip! I would like to check one thing: do I have to **(MUST)** use command `ern run-ios` to setup something or otherwise I won't be able to build the app? Because that's what it starts to look like...

What additional things `ern run-ios` does that hasn't been done beforehands by other commands e.g. `ern` (which seems to create ~/.ern folder), `ern create-container` (which seems to create ElectrodeContainer project under .ern) or `yarn` (which seems to download dependencies)?

Sorry for stupid questions, I'm just a native-only iOS developer suddenly involved in Electrode Native project. Learning new skills :)

**Comment by anilmanukonda** [comment]
@clairewli with your suggestions, I was able to get rid of those errors. but got some new ones related to duplicate declarations in ElectrodeContainer 

RN version 0.51

<img width="272" alt="rctduplicates" src="https://user-images.githubusercontent.com/8689237/36228337-d273d40c-1188-11e8-9db9-6278093ad0a6.png">

**Comment by clairewli** [comment]
@jomnius Thanks for raising the question. In fact, you don't have to run `ern run-ios` many times to run it again. It's a command we recommend for people we want just to have a try on our platform. Instead, you could use `ern create-container` Here's the document on the flags you can pass to create a container. I would recommend using this flag `--outDir/-o <directory>` so you will have your container generated. `ElectrodeContainer` by itself is a just a dynamic framework. You can add it as dependencies to your app as you normally will. Please let me know if that works for you or not.

**Comment by clairewli** [comment]
@anilmanukonda so here's what i did. `ern create-miniapp TestingMiniAPP`. then I did `ern run-ios` to build the project. For me, I'm using Xcode 9.2 and react 0.51 on ern v0.11.3. Now my simulator should pop out and showing `welcome to react native` screen. Then under my root dir for `TestingMiniApp`, I have a directory `ios` which contains my `ErnRunner.xcodeproj`. I open it and do a deep clean and delete DerivedData. Then I try to run the project, it failed with the crazy load of errors. Afterwards, I did `rm -rf ~/.rncache`. Run the project again, it builds the `Welcome to React Native` screen again.

**Comment by anilmanukonda** [comment]
@clairewli Right, I’m able to do all this as well. I don’t have a problem with ernrunner, my post is about generated container project and building that with a native project using carthage

**Comment by anilmanukonda** [comment]
I have cauldron configured in a repo and container being generated to a repo and that generated project has these errors.

**Comment by anilmanukonda** [comment]
@clairewli @belemaire any help on this?

**Comment by matheusalano** [comment]
I was getting the same kind of error every time I tried to build the container project with Carthage. The only way I found to fix my error was on Build Settings under ElectrodeContainer target. Find the `Header Search Paths` option and change everything (Debug, QA and Release) to `non-recursive`.
![captura de tela 2018-03-26 as 10 37 11](https://user-images.githubusercontent.com/11509362/37909575-acd6ba48-30e1-11e8-9f8b-46afcda873ed.png)

I hope it fixes your error too.
